include/seqan3/io/detail/chunking.hpp

```diff
@@ -0,0 +1,149 @@
+// ==========================================================================
+//                 SeqAn - The Library for Sequence Analysis
+// ==========================================================================
+// Copyright (c) 2006-2016, Knut Reinert, FU Berlin
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ==========================================================================
+
+#include <meta/meta.hpp>
+#include <range/v3/utility/nullptr_v.hpp>
+
+#include <range/v3/iterator_range.hpp>
+
+#include <seqan3/core/meta.hpp>
+
+#pragma once
+
+// namespace seqan3::detail
+// {
+//
+// template <typename t>
+// ranges::v3::iterator_range<t, t>
+// chunk_type_helper(t *);
+//
+// template <typename t>
+// using chunk_type_ = meta::_t<decltype(detail::chunk_type_helper(ranges::v3::_nullptr_v<t>()))>;
+// }  // namespace seqan3::detail
+
+// namespace seqan3
+// {
+// // Metafunction chunk_type
+// template <typename t>
+// struct chunk_type : meta::defer<detail::chunk_type_, std::remove_cv_t<t>>
+// {};
+//
+// template <typename t>
+// using chunk_type_t = typename chunk_type<t>::type;
+// }  // namespace seqan3
+
+namespace seqan3::detail
+{
+
+template <typename iterator_t>
+class chunk_decorator
```

Comment:
whats the point of this?

The general lack of documentation makes this very hard to understand.